src/OrleansRuntime/Messaging/OutboundMessageQueue.cs

```diff
@@ -125,6 +127,22 @@ public void SendMessage(Message msg)
                     return;
                 }
 
+                // check for simulation of lost messages
+                if (Silo.CurrentSilo.TestHookup.SimulatedMessageLoss != null)
```

Comment:
At the very least, encapsulate this logic completely in TestHookup, so that here you will only call:
if(Silo.CurrentSilo.TestHookup.ShouldDrop(msg)) messageCenter.SendRejection(msg.

But better would be just not to have this random error injection code here at all.
One time tests can be done on a private branch.
If you plan to have something like that run as a regular UnitTest in our test framework, than we can think together how to do that. It will also require being able to validate the result. But if it sis not meant to be an automated test, better not to pollute the code with it.